libbpf-cargo/src/gen.rs

```diff
@@ -143,12 +143,19 @@ fn get_prog_name(prog: *const libbpf_sys::bpf_program) -> Result<String> {
 }
 
 fn map_is_mmapable(map: *const libbpf_sys::bpf_map) -> bool {
-    let internal = unsafe { libbpf_sys::bpf_map__is_internal(map) };
     let def = unsafe { libbpf_sys::bpf_map__def(map) };
-    let mmapable = unsafe { (*def).map_flags } & libbpf_sys::BPF_F_MMAPABLE;
-    let name = get_map_name(map);
+    (unsafe { (*def).map_flags } & libbpf_sys::BPF_F_MMAPABLE) > 0
+}
+
+fn map_is_datasec(map: *const libbpf_sys::bpf_map) -> bool {
+    let internal = unsafe { libbpf_sys::bpf_map__is_internal(map) };
+    let mmapable = map_is_mmapable(map);
+    let has_datasec_name = match get_map_name(map).unwrap().as_ref().map(String::as_str) {
+        Some("rodata") | Some("data") | Some("bss") | Some("kconfig") => true,
+        _ => false,
+    };
 
-    internal && (mmapable > 0) && (name.is_ok() && name.unwrap().is_some())
+    internal && mmapable && has_datasec_name
```

Comment:
Let's not duplicate the hardcoded list in more than 1 place.

I think this can be simplified to:
```
let has_name = get_map_name(map).is_some();

internal && mmapable && has_name
```

Because if `internal` is true then a `Some(..)` returned by `get_map_name()` is already a valid internal map name

Reply:
Simplified